### PR TITLE
Search api

### PR DESCRIPTION
**Pull request by Rachana-Shetty011**
https://github.com/nandeeshkamath/youtube-tracker/issues/20

### CODE REVIEW

File: src/main/java/com/youtube/tracker/controller/TrackerController.java [hunk 1]
```diff
@@ -21,4 +29,12 @@ public String entryPoint() {
     public ResponseWrapper track() {
         return ResponseWrapper.success();
     }
+
+    @GetMapping("/search")
+    public ResponseWrapper<YoutubeSearchResponse> getChannelDetails(
+            @RequestParam @NotBlank String part,
+            @RequestParam @NotBlank String key,
+            @RequestParam @NotBlank String id) {
+        return ResponseWrapper.success(youtubeTrackerFeign.getChannels(part, id, key).getBody());
```
Comment (nandeeshkamath) [inline src/main/java/com/youtube/tracker/controller/TrackerController.java]: We don't need to expose this, we will just call POST /track which will call both youtube and telegram.
For this PR we do not need any changes in controller

File: build.gradle [hunk 2]
```diff
@@ -31,8 +31,9 @@ dependencies {
 	compileOnly 'org.projectlombok:lombok'
 	developmentOnly 'org.springframework.boot:spring-boot-devtools'
 	annotationProcessor 'org.projectlombok:lombok'
-
 	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	implementation 'org.jetbrains:annotations:15.0'
+	implementation 'org.springframework.boot:spring-boot-starter-validation'
```
Comment (nandeeshkamath) [inline build.gradle]: Do we need these?

File: src/main/java/com/youtube/tracker/controller/TrackerController.java (hunk 1 above)
Comment (Rachana-Shetty011) [reply src/main/java/com/youtube/tracker/controller/TrackerController.java:38]: ok
I'll remove

File: build.gradle (hunk 2 above)
Comment (Rachana-Shetty011) [reply build.gradle:36]: Had added it for validation purpose in controller, now will remove

File: src/main/java/com/youtube/tracker/feign/TelegramBotFeign.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import org.springframework.web.bind.annotation.RequestParam;
 
 // TODO: Add Configuration for server/client errors
-@FeignClient(value = "telegram-bot-feign", url = "{telegram-api.base-url}")
+@FeignClient(value = "telegram-bot-feign", url = "${telegram-api.base-url}")
```
Comment (nandeeshkamath) [inline src/main/java/com/youtube/tracker/feign/TelegramBotFeign.java:9]: Thanks for fixing!

**Review by nandeeshkamath — APPROVED** [review]

**Pull request merged**